counting_sort/counting_sort.c

```diff
@@ -0,0 +1,38 @@
+/* Following algorithm sorts the input array in ascending order
+*  Time complexity is O(n+k)
+*  Auxiliary space is O(n+k)
+*  n is number of elements and k is the range of input
+*  MAX is the maximum element in array
+*/
+
+#include <stdio.h>
+#include <string.h>
+
+void counting_sort(int arr[], int n, int MAX) {
+	// Function to sort input array arr.
+	int i;
+	int count[MAX + 1];  // Array to store count's of elements.
+	memset(count, 0, sizeof(count));
+	int temp[n];
+	for (i = 0; i < n; ++i)  // Count the occurences of contents of array.
+		count[arr[i]]++;
+	for (i = 1; i <= MAX; ++i)
+        count[i] += count[i-1];
+    for (i = 0; i < n; ++i) {  // Sort the array according to occurences of elements.
+        temp[count[arr[i]] - 1] = arr[i];
+        count[arr[i]]--;
+    }
+    for (i = 0; i < n; ++i)  // Copy the sorted array back to original array.
+        arr[i] = temp[i];
+}
+
+int main() {
+	int i;
+	int MAX = 10;  // MAX is the maximum number in the array
+	int arr[] = {10, 9, 8, 7, 6, 5, 4, 3, 2, 1};
+	int n = sizeof(arr) / sizeof(arr[0]);
+	counting_sort(arr, n, MAX);
```

Comment:
Full-uppercase variables in C are meant for constants. 
But this isn't a constant as the counting_sort function can be called many times in a program with different values of MAX. 
So please make it `max`.